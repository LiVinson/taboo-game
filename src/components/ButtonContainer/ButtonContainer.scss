.btnContainer {
    margin-top:1.5rem;
    display:flex;
    justify-content: space-between;
    
}