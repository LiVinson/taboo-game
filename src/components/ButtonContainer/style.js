import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`