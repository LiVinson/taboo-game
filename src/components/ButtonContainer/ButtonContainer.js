import React from "react"
import PropTypes from "prop-types"
import { Button, PrimaryButton, LargeButton } from "components/shared/Button"
import { StyledButtonContainer } from "./style"

//Appears at bottom of Taboo Card when a buttons prop is passed to taboo.
// Receives array of objects with button text and click handler
const ButtonContainer = ({ buttons }) => {
  return (
    <StyledButtonContainer>
      {buttons.map((buttonInfo, index, buttons) =>
        determineButtonType(buttonInfo, index, buttons.length)
      )}
    </StyledButtonContainer>
  )
}

ButtonContainer.propTypes = {
  buttons: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string.isRequired,
      onClick: PropTypes.func.IsRequired,
      type: PropTypes.string,
    })
  ),
}

/*3 possible buttons: regular, primary, and large regular
  When there are 2 or more buttons, first is always button (white) and second is primary button (color).
  If there are less than 2 buttons or more than 2 (e.g. 1 or 3 buttons) then that button is large button
  Margin is only added to large button when there are multiple buttons to separate it from buttons above
*/
const determineButtonType = (buttonInfo, index, length) => {

  if (index === 0 && length >= 2) {
    return <Button key={index} {...buttonInfo} />
  } else if (index === 1 && length >= 2) {
    return <PrimaryButton key={index} {...buttonInfo} />
  } else {
    return <LargeButton margin={length>2} key={index} {...buttonInfo} />
  }
}

export default ButtonContainer