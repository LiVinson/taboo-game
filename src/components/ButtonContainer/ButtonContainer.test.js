import React from "react"
import {mount} from "enzyme"
import renderer from "react-test-renderer"

import ButtonContainer from "./ButtonContainer"

it("renders correctly", () => {
    const props = {
        buttons: [{text: "Back", handleClick: jest.fn()}, {text: "Next", handleClick: jest.fn()}]
    }
    const ButtonContainerComponent = renderer.create(<ButtonContainer {...props}/>).toJSON();
 
    expect(ButtonContainerComponent).toMatchSnapshot()
})

it("receives a buttons array prop", () => {
    const props = {
        buttons: [{text: "Back", handleClick: jest.fn()}, {text: "Next", handleClick: jest.fn()}]
    }
    const ButtonContainerComponent = mount(<ButtonContainer {...props}/>)

    expect(ButtonContainerComponent.prop("buttons")).toHaveLength(2)
})