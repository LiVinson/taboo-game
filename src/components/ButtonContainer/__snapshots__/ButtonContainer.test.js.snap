// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="btnContainer"
>
  <button
    className="button "
    handleClick={[MockFunction]}
    text="Back"
  >
    Back
  </button>
  <button
    className="button "
    handleClick={[MockFunction]}
    text="Next"
  >
    Next
  </button>
</div>
`;
